mygit.rb

```diff
@@ -0,0 +1,365 @@
+require 'optparse'
+require 'optparse/time'
+require 'octokit'
+require 'io/console'
+require 'logger'
+
+$force = 'False'
+
+options ={
+:files => [],
+:repoandbranch => [],
+:force => 'False',
+:reviewers => []
+}
+
+OptionParser.new do |parser|
+   #Handling the -create Command
+
+   parser.on('-c', '--create=repo branch',Array, 'Create a Repo and Branch') do |create|
+        options[:repoandbranch] += create  
+        mygit_create(options[:repoandbranch])
+   end
+
+   #Handling the Check Status --status command
+
+   parser.on('-s','--status', 'Check Status of recent pull request and review request') do |status|
+        mygit_status()
+   end
+
+   #Handling the Git Configuration 
+
+   parser.on('--config', 'Configure Git') do
+        configExist?()
+   end
+
+   #Handling the Request Review Command --review
+   
+   parser.on('-r','--request = reviewer', Array, 'Request review from collaborators') do |reviewer|
+        mygit_request(options[:reviewers])
+   end
+
+   #Handling the prepare command
+   
+   parser.on('-p', '--prepare [master]', String , 'Rebase against Base') do |base|
+        options[:base]=base
+        if ARGV.include?('-f')
+                $force = 'True'
+        end
+        mygit_prepare(base)
+   end
+
+   #Handling the force command for push and Config
+
+   parser.on('-f', '--force', 'Force Push') do |force|
+        $force='True'
+   end
+
+
+options[:files] += ARGV
+options[:repoandbranch] += ARGV
+options[:reviewers] += ARGV
+
+$LOG = Logger.new('log_file.log', 'monthly')
+
+def configExist?()
+         $LOG.debug("<MyGit> : Checking if Config Exists")
+         repo = getreponame()
+         Dir.chdir "#{repo}/.#{repo}"
+         if File.file?('gitconfig.txt')
+                $LOG.debug("<MyGit> : <ConfigExist?> : The config file exists")
+                Dir.chdir('../..')
+                if ARGV.include?('-f')
+                        $LOG.debug("<MyGit> : <ConfigExist?> : Forcefully creating a new config")
+                        ConfigGit()
+                        return                        
+                end
+                puts "<MyGit> : Config Exists already \n use -f tag to force a new config"
+                $LOG.debug("<MyGit> : <ConfigExist?> : Config Already Exists!")
+         else
+                $LOG.debug("<MyGit> : <ConfigExist?> : Calling the ConfigGit function to configure the user's git")
+                Dir.chdir('../..')
```

Comment:
Why chdir ?

Reply:
I am using it to navigate between the git directory and the hidden directory I have created. The hidden directory contains all the files relevant to the application, like the git config, base branch etc..